code/datums/antagonists/datum_devil.dm

```diff
@@ -428,15 +404,49 @@ var/global/list/lawlorify = list (
 					H.forceMove(A)
 					A.oldform = H
 					A.set_name()
-					owner.transfer_to(A)
+					owner.mind.transfer_to(A)
 					if(SOULVALUE >= ARCH_THRESHOLD)
 						A.convert_to_archdevil()
 		else
 			throw EXCEPTION("Unable to find a blobstart landmark for hellish resurrection")
 	check_regression()
 
-/datum/devilinfo/proc/update_hud()
-	if(istype(owner.current, /mob/living/carbon))
-		var/mob/living/C = owner.current
+/datum/antagonist/devil/proc/update_hud()
+	if(istype(owner, /mob/living/))
```

Comment:
You don't need to check this, as the owner of an antag datum can only ever be living
You might need to check if(owner), though